### ISSUE TITLE
Teleport fails to start with teleport configure configuration in 7.1.1

### ISSUE THREAD

**Issue by stevenGravy** [gravitational/teleport#8206]
### Description

**What happened**:

Attempting to start a new cluster with a `teleport configure` generated `/etc/teleport.yaml` results in an error of `Proxy failed to establish connection to cluster`.  The server never starts properly.   This is not happening in 7.1.0 enterprise.

```bash
2021-09-12T22:53:45Z DEBU [AUTH:GRPC] Got heartbeat connection from 68a88076-2629-4aaf-b00f-97ad593c96cd.ds9.teleportdemo.com. auth/grpcserver.go:110
2021-09-12T22:53:50Z DEBU [AUTH:GRPC] Failed to receive heartbeat: rpc error: code = Canceled desc = context canceled auth/grpcserver.go:120
2021-09-12T22:53:50Z DEBU [AUTH:GRPC] Got heartbeat connection from 68a88076-2629-4aaf-b00f-97ad593c96cd.ds9.teleportdemo.com. auth/grpcserver.go:110
2021-09-12T22:53:55Z ERRO [PROC:1]    Proxy failed to establish connection to cluster: empty hostname. service/connect.go:67
2021-09-12T22:54:05Z ERRO [PROC:1]    Proxy failed to establish connection to cluster: empty hostname. service/connect.go:67
2021-09-12T22:54:15Z ERRO [PROC:1]    Proxy failed to establish connection to cluster: empty hostname. service/connect.go:67
```

**What you expected to happen**:

The same configuration file used with `7.1.0` works.

### Reproduction Steps
As minimally and precisely as possible, describe step-by-step how to reproduce the problem.
1. Install 7.1.1
2. Run the `teleport configure` command
```bash
 sudo teleport configure --acme --acme-email=user@mail.com --cluster-name=teleport.example.com
```
3. Run teleport start

### Server Details
- Teleport version (run `teleport version`): 7.1.1
- Server OS (e.g. from `/etc/os-release`): Ubuntu
- Where are you running Teleport? (e.g. AWS, GCP, Dedicated Hardware): AWS
- Additional details:

Example yaml:
```yaml
teleport:
  nodename: ip-172-31-28-206
  data_dir: /var/lib/teleport
  log:
    output: stderr
    severity: INFO
    format:
      output: text
  ca_pin: ""
  diag_addr: ""
auth_service:
  enabled: "yes"
  listen_addr: 0.0.0.0:3025
  cluster_name: ds9.myserver.com
  license_file: /var/lib/teleport/license.pem
ssh_service:
  enabled: "yes"
  labels:
    env: example
  commands:
  - name: hostname
    command: [hostname]
    period: 1m0s
proxy_service:
  enabled: "yes"
  listen_addr: 0.0.0.0:3023
  web_listen_addr: :443
  public_addr: ds9.myserver.com:443
  https_keypairs: []
  acme:
    enabled: "yes"
    email: steven@goteleport.com
```

**Comment by stevenGravy** [comment]
Setting the `web_listen_addr: 0.0.0.0:443` resolves it.

**Comment by webvictim** [comment]
I think this may have been broken since 7.0.0 as I've definitely seen that `:443` by itself before and not realised where it was coming from.

**Comment by r0mant** [comment]
The fix will be in 7.1.2.
